packages/compiler-cli/src/ngtsc/metadata/src/util.ts

```diff
@@ -93,7 +96,45 @@ export function extractDirectiveGuards(node: ClassDeclaration, reflector: Reflec
   const coercedInputFields =
       new Set(staticMembers.map(extractCoercedInput)
                   .filter((inputName): inputName is string => inputName !== null));
-  return {hasNgTemplateContextGuard, ngTemplateGuards, coercedInputFields};
+
+  const restrictedInputFields = new Set<string>();
+  const stringLiteralInputFields = new Set<string>();
+  const undeclaredInputFields = new Set<string>();
+
+  for (const fieldName of Object.keys(inputs)) {
+    const field = members.find(member => member.name === fieldName);
```

Comment:
Note to self: filter out getters to avoid being sensitive to ordering, i.e. a public getter and private setter must be recorded as restricted, regardless of whether the setter or getter is ordered first.

```suggestion
    const field = members.find(member => member.name === fieldName && member.kind !== ClassMemberKind.Getter);
```

Reply:
Alright, so I tried adding a test for this scenario and found out that a getter/setter pair needs to agree on a visibility, so this point is moot.